packages/headless/src/features/facets/facet-search-set/facet-search-reducer-helpers.ts

```diff
@@ -34,7 +41,11 @@ export function handleFacetSearchRegistration<T extends FacetSearchResponse>(
   }
 
   const isLoading = false;
-  const options = buildFacetSearchOptions(payload);
+  const baseOptions = {...defaultFacetSearchOptions, ...payload};
+  const options: FacetSearchStateOptions = {
+    ...baseOptions,
+    initialNumberOfValues: baseOptions.numberOfValues,
+  };
   const response = buildEmptyResponse();
 
   state[facetId] = {options, isLoading, response};
```

Comment:
Following on the comment above, the only change would be to add `initialNumberOfValues` here. No need to destructure base options,

```
const options =  buildFacetSearchOptions(payload);
const response = buildEmptyResponse();
const initialNumberOfValues = payload.numberOfValues;

state[facetId] = {options, isLoading, response, initialNumberOfValues}
```